### ISSUE TITLE
Add named exception to detect when a cluster node has been quarantined by others

### ISSUE THREAD

**Issue by garyiwu** [akka/akka#18758]
When a cluster node gets auto-downed/quarantined, it has to restart its own actor system before it can rejoin the cluster.  To do this we need a way to reliably detect when a node has been quarantined by other nodes.  

Currently, the only way to do this appears to be doing string matching on the exception message of an `AssociationError`, which is fragile.  Ideally there should be a named exception that we can check, so that the node can restart itself as appropriate.

**Comment by jdevelop** [comment]
:+1:

**Comment by jdevelop** [comment]
I faced the same problem and came out with the following solution:

- add the special case class **AssociationHandle.ActorSystemQuarantinedEvent** that will contain the local address of the system being quarantined, and remote address of the system that has quarantined the current one.
- upon receive of the quarantine message in the Endpoint - send a new **AssociationHandle.ActorSystemQuarantinedEvent** to the event stream of the current ActorSystem. 

Hence any actor could subscribe to those events and in case of quarantine - do something, e.g - trigger system restart.

I found this pretty flexible in terms of usage, however need somebody to review the approach and the proposed patch. Thanks.
